profile-lib/analyzer.rkt

```diff
@@ -122,23 +122,24 @@
 
 ;; returns a list of (cons item occurrences) for the items in l
 (define (get-counts l)
-  (let loop ([l l] [r '()])
-    (if (null? l)
-      r
-      (let ([1st (car l)])
-        (let loop* ([l1 '()] [c 1] [l (cdr l)])
-          (cond [(null? l) (loop l1 (cons (cons 1st c) r))]
-                [(eq? 1st (car l)) (loop* l1 (add1 c) (cdr l))]
-                [else (loop* (cons (car l) l1) c (cdr l))]))))))
+  (define h (make-hasheq))
+  (for ([x (in-list l)])
+    (hash-update! h x add1 0))
+  h)
 
 (module+ test
-  (check-equal? (get-counts '()) '())
-  (check-equal? (get-counts '(1)) '([1 . 1]))
-  (check-equal? (get-counts '(1 1 1)) '([1 . 3]))
-  (define (set=? xs ys) (null? (append (remove* xs ys) (remove* ys xs))))
-  (check set=? (get-counts '(1 2 3)) '([1 . 1] [2 . 1] [3 . 1]))
-  (check set=? (get-counts '(1 2 2 3 3 3)) '([1 . 1] [2 . 2] [3 . 3]))
-  (check set=? (get-counts '(3 1 2 3 2 3)) '([1 . 1] [2 . 2] [3 . 3])))
+  (define (hash=? a b)
```

Comment:
Why isn't this just `equal?`?

Reply:
Order of keys

Reply:
Oh I see—why not make the RHS a hash as well? Is that the question?

Reply:
Oh, I see. It's really dumb. You get this:

```
--------------------
FAILURE
name:       check
location:   analyzer.rkt:142:2
params:
  '(#<procedure:equal?>
  #hasheq((1 . 1) (2 . 2) (3 . 3))
  #hasheq((1 . 1) (2 . 2) (3 . 3)))
--------------------
```

It's something dumb about the keys not being `eq?` probably.

Reply:
Oh, right. Yes of course the keys are ints, they are immediates. Fixed.